src/contract.rs

```diff
@@ -39,17 +42,27 @@ pub trait TokenTrait {
     fn set_admin(e: Env, admin: Authorization, new_admin: Identifier);
 
     fn unfreeze(e: Env, admin: Authorization, id: Identifier);
+
+    fn decimals(e: Env) -> u32;
+
+    fn name(e: Env) -> Vec<u8>;
+
+    fn symbol(e: Env) -> Vec<u8>;
 }
 
 pub struct Token;
 
 #[contractimpl(export_if = "export")]
 impl TokenTrait for Token {
-    fn initialize(e: Env, admin: Identifier) {
+    fn initialize(e: Env, admin: Identifier, decimal: u32, name: Vec<u8>, symbol: Vec<u8>) {
```

Comment:
It occurred to me yesterday evening that this PR is using `Vec<u8>` to indicate strings, which is what is done in CAP-0054. But CAP-0054 is actually wrong in this regard (and many others...), and strings should be `Binary`.

Although this compiled when you wrote this code, it won't compile anymore after some fixes by @leighmcculloch. I'm going to update your branch, so it will show that the tests fail.

Reply:
Good point. Fixed.